tap_shiphero/sync.py

```diff
@@ -61,7 +49,19 @@ def sync_products(client, catalog, state, start_date, end_date, stream_id, strea
 
     write_schema(catalog, stream_id)
 
-    last_date = get_bookmark(state, stream_id, start_date)
+    last_date = bookmarks.get_bookmark(state, stream_id, 'datetime', start_date)
+
+    # Rip this out once all bookmarks are converted
+    old_style_bookmark = bookmarks.get_bookmark(state, stream_id, stream_id)
+    if old_style_bookmark:
+        # Use the old style bookmark
+        last_date = old_style_bookmark
+
+        # Write this bookmark in the new style
+        bookmarks.write_bookmark(state, stream_id, 'datetime', last_date)
+
+        # Purge the old style bookmark
+        bookmarks.clear_bookmark(state, stream_id, stream_id)
```

Comment:
Same here. Different structure, so I think this will leave it hanging. Just do `old_bookmark = state['bookmarks']['stream_id'].pop()` above.